src/mlpack/tests/main_tests/preprocess_imputer_test.cpp

```diff
@@ -42,53 +42,50 @@ TEST_CASE_METHOD(
   size_t inputSize = inputData.n_cols;
 
   // Input custom data points and labels.
-  SetInputParam("input_file", (std::string) "preprocess_imputer_test.csv");
+  SetInputParam("input", std::move(inputData));
   SetInputParam("missing_value", (std::string) "nan");
-  SetInputParam("output_file",
-      (std::string) "preprocess_imputer_output_test.csv");
+  REQUIRE(params.Get<arma::Row<size_t>>("output").n_cols == testSize);
 
   // Check for mean strategy.
   SetInputParam("strategy", (std::string) "mean");
 
   RUN_BINDING();
 
   // Now check that the output has desired dimensions.
-  data::Load(params.Get<std::string>("output_file"), outputData);
+  arma::mat outputData = params.Get<arma::mat>("output");
   REQUIRE(outputData.n_cols == inputSize);
   REQUIRE(outputData.n_rows == 3); // Input Dimension.
 
   // Reset passed strategy.
   ResetSettings();
 
   // Check for median strategy.
-  SetInputParam("input_file", (std::string) "preprocess_imputer_test.csv");
+  SetInputParam("input", std::move(inputData));
```

Comment:
Once you called `std::move(inputData)` earlier, `inputData` will now be empty.  So if you are going to use `inputData` multiple times in one test, just call `SetInputParam("input", inputData)` instead, so that `inputData` is copied instead of moved. :+1: